### PR TITLE
enable edma and sai support

### PR DESCRIPTION
**Pull request by hakehuang**
there is one issue in edma driver, the busy state judgement is wrong. provide a temp solution. still communite with owner for a solid soltuion.

### CODE REVIEW

**Comment by hakehuang** [comment]
@MaureenHelm , can you help to review it? it seems there is no setting in this repo to merge community pull request, Thanks

**Comment by hakehuang** [comment]
@MaureenHelm  for the edma driver bug fix, can we keep it in and them revert it when we rebase to next SDK release? This would be bridge several months gap I guess

File: mcux/drivers/imx/fsl_edma.c [hunk 1]
```diff
@@ -1058,10 +1058,18 @@ status_t EDMA_SubmitTransfer(edma_handle_t *handle, const edma_transfer_config_t
             this check mechanism, EDMA_CreatHandle will clear CSR register.
         */
         tmpCSR = tcdRegs->CSR;
+        #if 0
```
Comment (MaureenHelm) [inline mcux/drivers/imx/fsl_edma.c:1061]: remove the dead code block and replace it with the fix. Please also update `mcux/README` to mention the change

Comment (hakehuang) [reply mcux/drivers/imx/fsl_edma.c:1061]: ok

File: mcux/drivers/imx/CMakeLists.txt [hunk 2]
```diff
@@ -25,3 +25,7 @@ zephyr_sources_ifdef(CONFIG_ETH_MCUX          fsl_enet.c)
 zephyr_sources_ifdef(CONFIG_ENTROPY_MCUX_TRNG fsl_trng.c)
 zephyr_sources_ifdef(CONFIG_PWM_MCUX fsl_pwm.c)
 zephyr_sources_ifdef(CONFIG_VIDEO_MCUX_CSI    fsl_csi.c)
+zephyr_sources_ifdef(CONFIG_DMA_MCUX_EDMA  fsl_edma.c)
+zephyr_sources_ifdef(CONFIG_DMA_MCUX_EDMA  fsl_dmamux.c)
+zephyr_sources_ifdef(CONFIG_IMX_SAI        fsl_sai.c)
+zephyr_sources_ifdef(CONFIG_IMX_SAI_DMA    fsl_sai_edma.c)
```
Comment (MaureenHelm) [inline mcux/drivers/imx/CMakeLists.txt:31]: I don't remember seeing an SAI driver in your zephyr PR. Do you still need to build these files?

Comment (hakehuang) [reply mcux/drivers/imx/CMakeLists.txt:31]: my next plan is to enable SAI. @MaureenHelm.

Comment (MaureenHelm) [reply mcux/drivers/imx/CMakeLists.txt:31]: The driver should probably be called `CONFIG_I2S_MCUX_SAI`

Comment (hakehuang) [reply mcux/drivers/imx/CMakeLists.txt:31]: ok

File: mcux/drivers/imx/fsl_edma.c [hunk 3]
```diff
@@ -1058,7 +1058,7 @@ status_t EDMA_SubmitTransfer(edma_handle_t *handle, const edma_transfer_config_t
             this check mechanism, EDMA_CreatHandle will clear CSR register.
         */
         tmpCSR = tcdRegs->CSR;
-        if ((tmpCSR != 0UL) && ((tmpCSR & DMA_CSR_DONE_MASK) == 0UL))
+        if ((tmpCSR & DMA_CSR_ACTIVE_MASK) == DMA_CSR_ACTIVE_MASK)
```
Comment (MaureenHelm) [inline mcux/drivers/imx/fsl_edma.c:1061]: The same change is probably needed in `mcux/drivers/kinetis/fsl_edma.c` too?

Comment (hakehuang) [reply mcux/drivers/imx/fsl_edma.c:1061]: correct, the driver need update as well, the strange thing is thatin kinetis we does not have such issue. let me double check

**Comment by hakehuang** [comment]
> Please add a commit message body to `EDMA: fix a bug in dma channel active check`
> 
> Looks like you also need to rebase.

done

**Pull request merged**